skbio/metadata/_interval.py

```diff
@@ -90,6 +91,10 @@ def __init__(self, interval_metadata, locations,
 
     def _add(self):
         """Add the current ``Interval`` to the IntervalMetadata object."""
+        if self.locations[-1][-1] > self._interval_metadata.upper_bound:
```

Comment:
I added to check the lower bound in line 250. I also moved the check into `_locations_boundaries_setter` so the `locations` setter does not escape the check.

You were not talking about adding `lower_bound` as an attribute, right? Because in all of our use cases, it should always start with 0; otherwise, it needs other API change.